### DISCUSSION TITLE
GS has no way to suppress industry news messages

### DISCUSSION THREAD

**Discussion by andythenorth** [OpenTTD/OpenTTD#11706] [Features]
### Version of OpenTTD

13-ish

### Expected result

GS can add industries to the map without generating a lot of 'new industry' messages for players.

The specific case shown is at game start, with a GS that places industry, but this should be generally available to GS, as there may be other cases.  For example GS places many industries during gameplay, or GS wants to deliberately introduce an industry by stealth.  

Eh, I don't specify implementation, but possibly an optional parameter would work well for GSIndustryType.ProspectIndustry() and GSIndustryType.BuildIndustry().  Alternatively some modal thing where GS can enter a mode which suppresses all news generation for all commands?

I did consider whether grf should do this, by e.g. modifying the return result for callback 28, but I think it would be hard to achieve a good effect as grf won't know the intent of the GS.

### Actual result

Player sees many 'new industry' messages, after map

![image](https://user-images.githubusercontent.com/1780327/213908858-30d5ec6b-01ba-416c-a02e-6fce52f471ce.png)

### Steps to reproduce

I will provide a test GS later.

**Comment by 2TallTyler** [comment]
I'd agree with the implementation idea that GS should be able to specify "no message please" using an optional parameter.

**Comment by glx22** [comment]
Agreed for things happening at the end of mapgen, but during normal game loop messages should be created

**Comment by andythenorth** [comment]
Test implementation of optional suppression in https://github.com/andythenorth/OpenTTD/commit/8324d9abcd2354854c8fde062685bb4415e3f51a

This has issues:

- no backwards compatibility to existing scripts, currently they will just break on missing parameter
- AI might be able to suppress news messages about industries opening, which probably isn't desirable
- no update to AI/GS changelog

**Comment by 2TallTyler** [comment]
Regarding your test implementation, does script API support optional parameters? (`bool show_news = true`) If so, that could solve your first bullet point. I have no suggestions to prevent AI from secretly building industries. I can't imagine it would be a huge issue, but someone might complain. 🤷

**Reply by glx22** [reply]
Optional parameters are possible, but require low level handling (see ScriptVehicleList).
